diff --git lib/private/Federation/CloudIdManager.php lib/private/Federation/CloudIdManager.php
index b0996d5..d018b2b 100644
--- lib/private/Federation/CloudIdManager.php
+++ lib/private/Federation/CloudIdManager.php
@@ -71,7 +71,7 @@ class CloudIdManager implements ICloudIdManager {
 	 * @return ICloudId
 	 * @throws \InvalidArgumentException
 	 */
-	public function resolveCloudId(string $cloudId): ICloudId {
+	public function resolveCloudId(string $cloudId, ?string $displayName = null): ICloudId {
 		// TODO magic here to get the url and user instead of just splitting on @
 
 		foreach ($this->cloudIdResolvers as $resolver) {
@@ -112,7 +112,7 @@ class CloudIdManager implements ICloudIdManager {
 
 			if (!empty($user) && !empty($remote)) {
 				$remote = $this->ensureDefaultProtocol($remote);
-				return new CloudId($id, $user, $remote, null);
+				return new CloudId($id, $user, $remote, $displayName);
 			}
 		}
 		throw new \InvalidArgumentException('Invalid cloud id');
