diff --git lib/private/Federation/CloudIdManager.php lib/private/Federation/CloudIdManager.php
index f68eeed..2c73e2c 100644
--- lib/private/Federation/CloudIdManager.php
+++ lib/private/Federation/CloudIdManager.php
@@ -78,7 +78,7 @@ class CloudIdManager implements ICloudIdManager {
 	 * @return ICloudId
 	 * @throws \InvalidArgumentException
 	 */
-	public function resolveCloudId(string $cloudId): ICloudId {
+	public function resolveCloudId(string $cloudId, ?string $displayName = null): ICloudId {
 		// TODO magic here to get the url and user instead of just splitting on @
 
 		if (!$this->isValidCloudId($cloudId)) {
@@ -113,7 +113,7 @@ class CloudIdManager implements ICloudIdManager {
 
 			if (!empty($user) && !empty($remote)) {
 				$remote = $this->ensureDefaultProtocol($remote);
-				return new CloudId($id, $user, $remote, null);
+				return new CloudId($id, $user, $remote, $displayName);
 			}
 		}
 		throw new \InvalidArgumentException('Invalid cloud id');
