diff --git apps/federation/composer/composer/autoload_classmap.php apps/federation/composer/composer/autoload_classmap.php
index 1be343a..e63ace6 100644
--- apps/federation/composer/composer/autoload_classmap.php
+++ apps/federation/composer/composer/autoload_classmap.php
@@ -11,6 +11,9 @@ return array(
     'OCA\\Federation\\BackgroundJob\\GetSharedSecret' => $baseDir . '/../lib/BackgroundJob/GetSharedSecret.php',
     'OCA\\Federation\\BackgroundJob\\RequestSharedSecret' => $baseDir . '/../lib/BackgroundJob/RequestSharedSecret.php',
     'OCA\\Federation\\Command\\SyncFederationAddressBooks' => $baseDir . '/../lib/Command/SyncFederationAddressBooks.php',
+	'OCA\\Federation\\Command\\AddTrustedServer' => $baseDir . '/../lib/Command/AddTrustedServer.php',
+	'OCA\\Federation\\Command\\ListTrustedServer' => $baseDir . '/../lib/Command/ListTrustedServer.php',
+	'OCA\\Federation\\Command\\RemoveTrustedServer' => $baseDir . '/../lib/Command/RemoveTrustedServer.php',
     'OCA\\Federation\\Controller\\OCSAuthAPIController' => $baseDir . '/../lib/Controller/OCSAuthAPIController.php',
     'OCA\\Federation\\Controller\\SettingsController' => $baseDir . '/../lib/Controller/SettingsController.php',
     'OCA\\Federation\\DAV\\FedAuth' => $baseDir . '/../lib/DAV/FedAuth.php',
