diff --git apps/files_sharing/lib/Controller/ShareController.php apps/files_sharing/lib/Controller/ShareController.php
index bc078cb..dd58718 100644
--- apps/files_sharing/lib/Controller/ShareController.php
+++ apps/files_sharing/lib/Controller/ShareController.php
@@ -313,6 +313,9 @@ class ShareController extends AuthPublicShareController {
 			throw new NotFoundException($this->l10n->t('This share does not exist or is no longer available'));
 		}
 
+		if ($this->config->getSystemValueBool('iserv_disable_file_downloads', false)) {
+			$share->setHideDownload(true);
+		}
 		$shareNode = $share->getNode();
 
 		try {
