diff --git a/apps/files_sharing/src/views/SharingDetailsTab.vue b/apps/files_sharing/src/views/SharingDetailsTab.vue
index 67bfac3..5cb49a3 100644
--- a/apps/files_sharing/src/views/SharingDetailsTab.vue
+++ b/apps/files_sharing/src/views/SharingDetailsTab.vue
@@ -28,7 +28,8 @@
 						<ViewIcon :size="20" />
 						<span>{{ t('files_sharing', 'View only') }}</span>
 					</NcCheckboxRadioSwitch>
-					<NcCheckboxRadioSwitch :button-variant="true"
+					<NcCheckboxRadioSwitch v-if="canEditFileType && !isPublicShare"
+                        :button-variant="true"
 						:checked.sync="sharingPermission"
 						:value="bundledPermissions.ALL.toString()"
 						name="sharing_permission_radio"
@@ -36,10 +37,7 @@
 						button-variant-grouped="vertical"
 						@update:checked="toggleCustomPermissions">
 						<EditIcon :size="20" />
-						<template v-if="allowsFileDrop">
-							<span>{{ t('files_sharing', 'Allow upload and editing') }}</span>
-						</template>
-						<template v-else>
+						<template>
 							<span>{{ t('files_sharing', 'Allow editing') }}</span>
 						</template>
 					</NcCheckboxRadioSwitch>
@@ -146,10 +144,10 @@
 						</label>
 						<textarea id="share-note-textarea" :value="share.note" @input="share.note = $event.target.value" />
 					</template>
-					<NcCheckboxRadioSwitch :checked.sync="setCustomPermissions">
+					<NcCheckboxRadioSwitch v-if="!isPublicShare" :checked.sync="setCustomPermissions">
 						{{ t('files_sharing', 'Custom permissions') }}
 					</NcCheckboxRadioSwitch>
-					<section v-if="setCustomPermissions" class="custom-permissions-group">
+					<section v-if="setCustomPermissions && !isPublicShare" class="custom-permissions-group">
 						<NcCheckboxRadioSwitch :disabled="!allowsFileDrop && share.type === SHARE_TYPES.SHARE_TYPE_LINK"
 							:checked.sync="hasRead">
 							{{ t('files_sharing', 'Read') }}
@@ -157,7 +155,7 @@
 						<NcCheckboxRadioSwitch v-if="isFolder" :disabled="!canSetCreate" :checked.sync="canCreate">
 							{{ t('files_sharing', 'Create') }}
 						</NcCheckboxRadioSwitch>
-						<NcCheckboxRadioSwitch :disabled="!canSetEdit" :checked.sync="canEdit">
+						<NcCheckboxRadioSwitch :disabled="!canSetEdit || !canEditFileType" :checked.sync="canEdit">
 							{{ t('files_sharing', 'Update') }}
 						</NcCheckboxRadioSwitch>
 						<NcCheckboxRadioSwitch v-if="config.isResharingAllowed && share.type !== SHARE_TYPES.SHARE_TYPE_LINK"
@@ -284,6 +282,35 @@ export default {
 	},

 	computed: {
+        canEditFileType () {
+            // Taken from https://github.com/nextcloud/text/blob/0320046ffe8b853e8e329a9b865cdf3633228509/lib/DirectEditing/TextDirectEditor.php#L83
+            const supportedMimeTypes = [
+                'text/markdown',
+                'text/plain',
+                'application/cmd',
+                'application/x-empty',
+                'application/x-msdos-program',
+                'application/javascript',
+                'application/json',
+                'application/x-perl',
+                'application/x-php',
+                'application/x-tex',
+                'application/xml',
+                'application/yaml',
+                'text/css',
+                'text/csv',
+                'text/html',
+                'text/org',
+                'text/x-c',
+                'text/x-c++src',
+                'text/x-h',
+                'text/x-java-source',
+                'text/x-ldif',
+                'text/x-python',
+                'text/x-shellscript'
+            ];
+            return this.isFolder || supportedMimeTypes.includes(this.fileInfo.mimetype?.toLowerCase());
+        },
 		title() {
 			let title = t('files_sharing', 'Share with ')
 			if (this.share.type === this.SHARE_TYPES.SHARE_TYPE_USER) {
