Index: apps/files_sharing/src/components/SharingEntryLink.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apps/files_sharing/src/components/SharingEntryLink.vue b/apps/files_sharing/src/components/SharingEntryLink.vue
--- a/apps/files_sharing/src/components/SharingEntryLink.vue	(revision 1debe060a87e1d50e18dd47b29ea32de3dc4bac6)
+++ b/apps/files_sharing/src/components/SharingEntryLink.vue	(date 1665043804574)
@@ -178,7 +178,7 @@
 					</template>
 
 					<!-- file -->
-					<ActionCheckbox v-if="!isFolder"
+					<ActionCheckbox v-if="!isFolder && canEditFileType"
 						:checked.sync="canUpdate"
 						:disabled="saving"
 						@change="queueUpdate('permissions')">
@@ -600,6 +600,36 @@
 			return this.fileInfo.type === 'dir'
 		},
 
+		canEditFileType() {
+			// Taken from https://github.com/nextcloud/text/blob/0320046ffe8b853e8e329a9b865cdf3633228509/lib/DirectEditing/TextDirectEditor.php#L83
+			const supportedMimeTypes = [
+				'text/markdown',
+				'text/plain',
+				'application/cmd',
+				'application/x-empty',
+				'application/x-msdos-program',
+				'application/javascript',
+				'application/json',
+				'application/x-perl',
+				'application/x-php',
+				'application/x-tex',
+				'application/xml',
+				'application/yaml',
+				'text/css',
+				'text/csv',
+				'text/html',
+				'text/org',
+				'text/x-c',
+				'text/x-c++src',
+				'text/x-h',
+				'text/x-java-source',
+				'text/x-ldif',
+				'text/x-python',
+				'text/x-shellscript'
+			];
+			return supportedMimeTypes.includes(this.fileInfo.mimetype?.toLowerCase());
+		},
+
 		/**
 		 * Does the current file/folder have create permissions
 		 * TODO: move to a proper FileInfo model?
