diff --git apps/settings/lib/Settings/Personal/Security/Authtokens.php apps/settings/lib/Settings/Personal/Security/Authtokens.php
index e0509b2..a6b293b 100644
--- apps/settings/lib/Settings/Personal/Security/Authtokens.php
+++ apps/settings/lib/Settings/Personal/Security/Authtokens.php
@@ -14,6 +14,7 @@ use OC\Authentication\Token\IToken;
 use OCP\AppFramework\Http\TemplateResponse;
 use OCP\AppFramework\Services\IInitialState;
 use OCP\Authentication\Exceptions\InvalidTokenException;
+use OCP\IConfig;
 use OCP\ISession;
 use OCP\IUserSession;
 use OCP\Session\Exceptions\SessionNotAvailableException;
@@ -24,6 +25,7 @@ class Authtokens implements ISettings {
 
 	public function __construct(
 		private IAuthTokenProvider $tokenProvider,
+		private IConfig $config,
 		private ISession $session,
 		private IUserSession $userSession,
 		private IInitialState $initialState,
@@ -39,7 +41,7 @@ class Authtokens implements ISettings {
 
 		$this->initialState->provideInitialState(
 			'can_create_app_token',
-			$this->userSession->getImpersonatingUserID() === null
+			($this->userSession->getImpersonatingUserID() === null) && !($this->config->getSystemValueBool('iserv_disable_app_passwords'))
 		);
 
 		return new TemplateResponse('settings', 'settings/personal/security/authtokens');
