diff --git apps/theming/src/UserThemes.vue apps/theming/src/UserThemes.vue
index 75d8896..da9fcc7 100644
--- apps/theming/src/UserThemes.vue
+++ apps/theming/src/UserThemes.vue
@@ -26,10 +26,6 @@
 		<NcSettingsSection :name="t('theming', 'Appearance and accessibility settings')"
 			:limit-width="false"
 			class="theming">
-			<!-- eslint-disable-next-line vue/no-v-html -->
-			<p v-html="description" />
-			<!-- eslint-disable-next-line vue/no-v-html -->
-			<p v-html="descriptionDetail" />
 
 			<div class="theming__preview-list">
 				<ItemPreview v-for="theme in themes"
@@ -61,18 +57,6 @@
 			</NcCheckboxRadioSwitch>
 		</NcSettingsSection>
 
-		<NcSettingsSection :name="t('theming', 'Background')"
-			class="background"
-			data-user-theming-background-disabled>
-			<template v-if="isUserThemingDisabled">
-				<p>{{ t('theming', 'Customization has been disabled by your administrator') }}</p>
-			</template>
-			<template v-else>
-				<p>{{ t('theming', 'Set a custom background') }}</p>
-				<BackgroundSettings class="background__grid" @update:background="refreshGlobalStyles" />
-			</template>
-		</NcSettingsSection>
-
 		<NcSettingsSection :name="t('theming', 'Keyboard shortcuts')">
 			<p>{{ t('theming', 'In some cases keyboard shortcuts can interfere with accessibility tools. In order to allow focusing on your tool correctly you can disable all keyboard shortcuts here. This will also disable all available shortcuts in apps.') }}</p>
 			<NcCheckboxRadioSwitch class="theming__preview-toggle"
@@ -94,7 +78,6 @@ import axios from '@nextcloud/axios'
 import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
 import NcSettingsSection from '@nextcloud/vue/dist/Components/NcSettingsSection.js'
 
-import BackgroundSettings from './components/BackgroundSettings.vue'
 import ItemPreview from './components/ItemPreview.vue'
 import UserAppMenuSection from './components/UserAppMenuSection.vue'
 
@@ -112,7 +95,6 @@ export default {
 		ItemPreview,
 		NcCheckboxRadioSwitch,
 		NcSettingsSection,
-		BackgroundSettings,
 		UserAppMenuSection,
 	},
 
