diff --git apps/theming/src/UserTheming.vue apps/theming/src/UserTheming.vue
index 4766886..15c41b0 100644
--- apps/theming/src/UserTheming.vue
+++ apps/theming/src/UserTheming.vue
@@ -7,10 +7,6 @@
 	<section>
 		<NcSettingsSection :name="t('theming', 'Appearance and accessibility settings')"
 			class="theming">
-			<!-- eslint-disable-next-line vue/no-v-html -->
-			<p v-html="description" />
-			<!-- eslint-disable-next-line vue/no-v-html -->
-			<p v-html="descriptionDetail" />
 
 			<div class="theming__preview-list">
 				<ItemPreview v-for="theme in themes"
@@ -51,16 +47,6 @@
 				@refresh-styles="refreshGlobalStyles" />
 		</NcSettingsSection>
 
-		<NcSettingsSection class="background"
-			:name="t('theming', 'Background and color')"
-			:description="isUserThemingDisabled
-				? t('theming', 'Customization has been disabled by your administrator')
-				: t('theming', 'The background can be set to an image from the default set, a custom uploaded image, or a plain color.')">
-			<BackgroundSettings v-if="!isUserThemingDisabled"
-				class="background__grid"
-				@update:background="refreshGlobalStyles" />
-		</NcSettingsSection>
-
 		<NcSettingsSection :name="t('theming', 'Keyboard shortcuts')"
 			:description="t('theming', 'In some cases keyboard shortcuts can interfere with accessibility tools. In order to allow focusing on your tool correctly you can disable all keyboard shortcuts here. This will also disable all available shortcuts in apps.')">
 			<NcCheckboxRadioSwitch class="theming__preview-toggle"
@@ -85,7 +71,6 @@ import axios from '@nextcloud/axios'
 import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
 import NcSettingsSection from '@nextcloud/vue/dist/Components/NcSettingsSection.js'
 
-import BackgroundSettings from './components/BackgroundSettings.vue'
 import ItemPreview from './components/ItemPreview.vue'
 import UserAppMenuSection from './components/UserAppMenuSection.vue'
 import UserPrimaryColor from './components/UserPrimaryColor.vue'
@@ -104,7 +89,6 @@ export default {
 		ItemPreview,
 		NcCheckboxRadioSwitch,
 		NcSettingsSection,
-		BackgroundSettings,
 		UserAppMenuSection,
 		UserPrimaryColor,
 	},
@@ -135,38 +119,6 @@ export default {
 		selectedTheme() {
 			return this.themes.find(theme => theme.enabled === true) || this.themes[0]
 		},
-
-		description() {
-			// using the `t` replace method escape html, we have to do it manually :/
-			return t(
-				'theming',
-				'Universal access is very important to us. We follow web standards and check to make everything usable also without mouse, and assistive software such as screenreaders. We aim to be compliant with the {guidelines}Web Content Accessibility Guidelines{linkend} 2.1 on AA level, with the high contrast theme even on AAA level.',
-			)
-				.replace('{guidelines}', this.guidelinesLink)
-				.replace('{linkend}', '</a>')
-		},
-
-		guidelinesLink() {
-			return '<a target="_blank" href="https://www.w3.org/WAI/standards-guidelines/wcag/" rel="noreferrer nofollow">'
-		},
-
-		descriptionDetail() {
-			return t(
-				'theming',
-				'If you find any issues, do not hesitate to report them on {issuetracker}our issue tracker{linkend}. And if you want to get involved, come join {designteam}our design team{linkend}!',
-			)
-				.replace('{issuetracker}', this.issuetrackerLink)
-				.replace('{designteam}', this.designteamLink)
-				.replace(/\{linkend\}/g, '</a>')
-		},
-
-		issuetrackerLink() {
-			return '<a target="_blank" href="https://github.com/nextcloud/server/issues/" rel="noreferrer nofollow">'
-		},
-
-		designteamLink() {
-			return '<a target="_blank" href="https://nextcloud.com/design" rel="noreferrer nofollow">'
-		},
 	},
 
 	watch: {
