diff --git apps/richdocuments/src/document.js apps/richdocuments/src/document.js
index 17f0c0cb4..eeaf4021b 100644
--- apps/richdocuments/src/document.js
+++ apps/richdocuments/src/document.js
@@ -167,8 +167,6 @@ const documentsMain = {
 				+ '<input name="ui_defaults" value="' + getUIDefaults() + '" type="hidden"/>'
 				+ '<input name="css_variables" value="' + generateCSSVarTokens() + '" type="hidden"/>'
 				+ '<input name="theme" value="' + getCollaboraTheme() + '" type="hidden"/>'
-				// buy product for new customer users
-				+ '<input name="buy_product" value="https://nextcloud.com/pricing" type="hidden"/>'
 				+ '<input name="host_session_id" value="nextcloud ' + OC.config.version
 					+ ' - richdocuments ' + getCapabilities().version + '" type="hidden"/>'
 				+ '</form>'
@@ -229,8 +227,6 @@ const documentsMain = {
 				+ '<input name="ui_defaults" value="' + getUIDefaults() + '" type="hidden"/>'
 				+ '<input name="css_variables" value="' + generateCSSVarTokens() + '" type="hidden"/>'
 				+ '<input name="theme" value="' + getCollaboraTheme() + '" type="hidden"/>'
-				// buy product for new customer users
-				+ '<input name="buy_product" value="https://nextcloud.com/pricing" type="hidden"/>'
 				+ '<input name="host_session_id" value="nextcloud ' + OC.config.version
 					+ ' - richdocuments ' + getCapabilities().version + '" type="hidden"/>'
 				+ '</form>'
