diff --git apps/federation/lib/BackgroundJob/GetSharedSecret.php apps/federation/lib/BackgroundJob/GetSharedSecret.php
index 859b62d..82c442d 100644
--- apps/federation/lib/BackgroundJob/GetSharedSecret.php
+++ apps/federation/lib/BackgroundJob/GetSharedSecret.php
@@ -92,7 +92,8 @@ class GetSharedSecret extends Job {
 		if ($created < $deadline) {
 			$this->logger->warning("The job to get the shared secret job is too old and gets stopped now without retention. Setting server status of '{$target}' to failure.");
 			$this->retainJob = false;
-			$this->trustedServers->setServerStatus($target, TrustedServers::STATUS_FAILURE);
+			/* Users cannot manually trigger a retry, therefore always retry automatically. */
+			$this->trustedServers->setServerStatus($target, TrustedServers::STATUS_ACCESS_REVOKED);
 			return;
 		}
 
