diff --git apps/federation/lib/BackgroundJob/RequestSharedSecret.php apps/federation/lib/BackgroundJob/RequestSharedSecret.php
index e89e5ff..2fb0c3e 100644
--- apps/federation/lib/BackgroundJob/RequestSharedSecret.php
+++ apps/federation/lib/BackgroundJob/RequestSharedSecret.php
@@ -34,6 +34,7 @@ namespace OCA\Federation\BackgroundJob;
 
 use GuzzleHttp\Exception\ClientException;
 use GuzzleHttp\Exception\RequestException;
+use OCA\Federation\DbHandler;
 use OCA\Federation\TrustedServers;
 use OCP\AppFramework\Http;
 use OCP\AppFramework\Utility\ITimeFactory;
@@ -70,6 +71,7 @@ class RequestSharedSecret extends Job {
 		private IDiscoveryService $ocsDiscoveryService,
 		private LoggerInterface $logger,
 		ITimeFactory $timeFactory,
+		private DbHandler $dbHandler,
 	) {
 		parent::__construct($timeFactory);
 		$this->httpClient = $httpClientService->newClient();
@@ -176,15 +178,18 @@ class RequestSharedSecret extends Job {
 		$token = $argument['token'];
 		$attempt = $this->getAttempt($argument) + 1;
 
-		$this->jobList->add(
-			RequestSharedSecret::class,
-			[
-				'url' => $url,
-				'token' => $token,
-				'created' => $created,
-				'attempt' => $attempt
-			]
-		);
+		$dbToken =$this->dbHandler->getToken($url);
+		if (hash_equals($dbToken, $token)) {
+			$this->jobList->add(
+				RequestSharedSecret::class,
+				[
+					'url' => $url,
+					'token' => $token,
+					'created' => $created,
+					'attempt' => $attempt
+				]
+			);
+		}
 	}
 
 	protected function getAttempt(array $argument): int {
