diff --git lib/private/Collaboration/Collaborators/RemotePlugin.php lib/private/Collaboration/Collaborators/RemotePlugin.php
index afa214a..c30f0fc 100644
--- lib/private/Collaboration/Collaborators/RemotePlugin.php
+++ lib/private/Collaboration/Collaborators/RemotePlugin.php
@@ -65,7 +65,7 @@ class RemotePlugin implements ISearchPlugin {
 						$cloudIdType = $cloudIdData['type'];
 					}
 					try {
-						[$remoteUser, $serverUrl] = $this->splitUserRemote($cloudId);
+						[$remoteUser, $serverUrl] = $this->splitUserRemote($cloudId, $contact['FN'] ?? null);
 					} catch (\InvalidArgumentException $e) {
 						continue;
 					}
@@ -108,6 +108,7 @@ class RemotePlugin implements ISearchPlugin {
 								'server' => $serverUrl,
 								'isTrustedServer' => $this->trustedServers?->isTrustedServer($serverUrl) ?? false,
 							],
+							'shareWithDisplayNameUnique' => is_array($contact['EMAIL']) ? array_values($contact['EMAIL'])[0] ?? '': $contact['EMAIL'] ?? '',
 						];
 					} else {
 						$result['wide'][] = [
@@ -121,6 +122,7 @@ class RemotePlugin implements ISearchPlugin {
 								'server' => $serverUrl,
 								'isTrustedServer' => $this->trustedServers?->isTrustedServer($serverUrl) ?? false,
 							],
+							'shareWithDisplayNameUnique' => is_array($contact['EMAIL']) ? array_values($contact['EMAIL'])[0] ?? '' : $contact['EMAIL'] ?? '',
 						];
 					}
 				}
@@ -166,9 +168,9 @@ class RemotePlugin implements ISearchPlugin {
 	 * @return array [user, remoteURL]
 	 * @throws \InvalidArgumentException
 	 */
-	public function splitUserRemote(string $address): array {
+	public function splitUserRemote(string $address, ?string $displayName = null): array {
 		try {
-			$cloudId = $this->cloudIdManager->resolveCloudId($address);
+			$cloudId = $this->cloudIdManager->resolveCloudId($address, $displayName);
 			return [$cloudId->getUser(), $this->cloudIdManager->removeProtocolFromUrl($cloudId->getRemote(), true)];
 		} catch (\InvalidArgumentException $e) {
 			throw new \InvalidArgumentException('Invalid Federated Cloud ID', 0, $e);
