diff --git lib/private/Collaboration/Collaborators/RemotePlugin.php lib/private/Collaboration/Collaborators/RemotePlugin.php
index f226715..22c1fec 100644
--- lib/private/Collaboration/Collaborators/RemotePlugin.php
+++ lib/private/Collaboration/Collaborators/RemotePlugin.php
@@ -66,7 +66,7 @@ class RemotePlugin implements ISearchPlugin {
 						$cloudIdType = $cloudIdData['type'];
 					}
 					try {
-						[$remoteUser, $serverUrl] = $this->splitUserRemote($cloudId);
+						[$remoteUser, $serverUrl] = $this->splitUserRemote($cloudId, $contact['FN'] ?? null);
 					} catch (\InvalidArgumentException $e) {
 						continue;
 					}
@@ -109,6 +109,7 @@ class RemotePlugin implements ISearchPlugin {
 								'server' => $serverUrl,
 								'isTrustedServer' => $this->trustedServers?->isTrustedServer($serverUrl) ?? false,
 							],
+							'shareWithDisplayNameUnique' => is_array($contact['EMAIL']) ? array_values($contact['EMAIL'])[0] ?? '': $contact['EMAIL'] ?? '',
 						];
 					} else {
 						$result['wide'][] = [
@@ -122,6 +123,7 @@ class RemotePlugin implements ISearchPlugin {
 								'server' => $serverUrl,
 								'isTrustedServer' => $this->trustedServers?->isTrustedServer($serverUrl) ?? false,
 							],
+							'shareWithDisplayNameUnique' => is_array($contact['EMAIL']) ? array_values($contact['EMAIL'])[0] ?? '' : $contact['EMAIL'] ?? '',
 						];
 					}
 				}
@@ -167,9 +169,9 @@ class RemotePlugin implements ISearchPlugin {
 	 * @return array [user, remoteURL]
 	 * @throws \InvalidArgumentException
 	 */
-	public function splitUserRemote(string $address): array {
+	public function splitUserRemote(string $address, ?string $displayName = null): array {
 		try {
-			$cloudId = $this->cloudIdManager->resolveCloudId($address);
+			$cloudId = $this->cloudIdManager->resolveCloudId($address, $displayName);
 			return [$cloudId->getUser(), $this->cloudIdManager->removeProtocolFromUrl($cloudId->getRemote(), true)];
 		} catch (\InvalidArgumentException $e) {
 			throw new \InvalidArgumentException('Invalid Federated Cloud ID', 0, $e);
