diff --git apps/richdocuments/lib/Service/CapabilitiesService.php apps/richdocuments/lib/Service/CapabilitiesService.php
index e66ce2321..473d33cf7 100644
--- apps/richdocuments/lib/Service/CapabilitiesService.php
+++ apps/richdocuments/lib/Service/CapabilitiesService.php
@@ -106,7 +106,7 @@ class CapabilitiesService extends CachedRequestService {
 			return $this->capabilitites['productName'];
 		}
 
-		return $this->l10n->t('Nextcloud Office');
+		return $this->l10n->t('IServ Office');
 	}
 
 	public function hasOtherOOXMLApps(): bool {
