diff --git apps/richdocuments/lib/Service/CapabilitiesService.php apps/richdocuments/lib/Service/CapabilitiesService.php
index 60a87a731..d7d5f3a17 100644
--- apps/richdocuments/lib/Service/CapabilitiesService.php
+++ apps/richdocuments/lib/Service/CapabilitiesService.php
@@ -115,7 +115,7 @@ class CapabilitiesService extends CachedRequestService {
 			return $this->capabilitites['productName'];
 		}
 
-		return $this->l10n->t('Nextcloud Office');
+		return $this->l10n->t('IServ Office');
 	}
 
 	public function hasOtherOOXMLApps(): bool {
