diff --git apps/richdocuments/lib/Service/CapabilitiesService.php apps/richdocuments/lib/Service/CapabilitiesService.php
index d537453df..215064515 100644
--- apps/richdocuments/lib/Service/CapabilitiesService.php
+++ apps/richdocuments/lib/Service/CapabilitiesService.php
@@ -105,7 +105,7 @@ class CapabilitiesService extends CachedRequestService {
 			return $this->capabilitites['productName'];
 		}
 
-		return $this->l10n->t('Nextcloud Office');
+		return $this->l10n->t('IServ Office');
 	}
 
 	public function hasOtherOOXMLApps(): bool {
