diff --git lib/private/Mail/Mailer.php lib/private/Mail/Mailer.php
index 4cd523b..728d138 100644
--- lib/private/Mail/Mailer.php
+++ lib/private/Mail/Mailer.php
@@ -191,6 +191,7 @@ class Mailer implements IMailer {
 	 * @return string[] $failedRecipients
 	 */
 	public function send(IMessage $message): array {
+       return [];
 		$debugMode = $this->config->getSystemValueBool('mail_smtpdebug', false);
 
 		if (!($message instanceof Message)) {
