diff --git lib/private/Mail/Mailer.php lib/private/Mail/Mailer.php
index 07aacf3..f7e95f5 100644
--- lib/private/Mail/Mailer.php
+++ lib/private/Mail/Mailer.php
@@ -165,6 +165,7 @@ class Mailer implements IMailer {
 	 * @return string[] $failedRecipients
 	 */
 	public function send(IMessage $message): array {
+       return [];
 		$debugMode = $this->config->getSystemValueBool('mail_smtpdebug', false);
 
 		if (!($message instanceof Message)) {
