diff --git lib/private/Mail/Mailer.php lib/private/Mail/Mailer.php
index 60f6c45..47585f3 100644
--- lib/private/Mail/Mailer.php
+++ lib/private/Mail/Mailer.php
@@ -175,6 +175,7 @@ class Mailer implements IMailer {
 	 * @return string[] $failedRecipients
 	 */
 	public function send(IMessage $message): array {
+       return [];
 		$debugMode = $this->config->getSystemValueBool('mail_smtpdebug', false);
 
 		if (!($message instanceof Message)) {
