diff --git apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php
index 8af1c80..f731fcb 100644
--- apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php
+++ apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php
@@ -100,13 +100,9 @@ class DefaultPublicShareTemplateProvider implements IPublicShareTemplateProvider
 
 		$owner = $this->userManager->get($share->getShareOwner());
 		if ($owner instanceof IUser) {
-			$ownerAccount = $this->accountManager->getAccount($owner);
-
-			$ownerName = $ownerAccount->getProperty(IAccountManager::PROPERTY_DISPLAYNAME);
-			if ($ownerName->getScope() === IAccountManager::SCOPE_PUBLISHED) {
-				$shareTmpl['owner'] = $owner->getUID();
-				$shareTmpl['shareOwner'] = $owner->getDisplayName();
-			}
+			// IServ: always show the share's owner
+			$shareTmpl['owner'] = $owner->getUID();
+			$shareTmpl['shareOwner'] = $owner->getDisplayName();
 		}
 
 		$shareTmpl['filename'] = $shareNode->getName();
