diff --git apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php
index b6644f1..8eff1c7 100644
--- apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php
+++ apps/files_sharing/lib/DefaultPublicShareTemplateProvider.php
@@ -68,15 +68,11 @@ class DefaultPublicShareTemplateProvider implements IPublicShareTemplateProvider
 		// Only make the share owner public if they allowed to show their name
 		$owner = $this->userManager->get($share->getShareOwner());
 		if ($owner instanceof IUser) {
-			$ownerAccount = $this->accountManager->getAccount($owner);
-
-			$ownerNameProperty = $ownerAccount->getProperty(IAccountManager::PROPERTY_DISPLAYNAME);
-			if ($ownerNameProperty->getScope() === IAccountManager::SCOPE_PUBLISHED) {
-				$ownerId = $owner->getUID();
-				$ownerName = $owner->getDisplayName();
-				$this->initialState->provideInitialState('owner', $ownerId);
-				$this->initialState->provideInitialState('ownerDisplayName', $ownerName);
-			}
+			// IServ: always show the share's owner
+			$ownerId = $owner->getUID();
+			$ownerName = $owner->getDisplayName();
+			$this->initialState->provideInitialState('owner', $ownerId);
+			$this->initialState->provideInitialState('ownerDisplayName', $ownerName);
 		}
 
 		$view = 'public-share';
