diff --git core/src/components/setup/RecommendedApps.vue core/src/components/setup/RecommendedApps.vue
index d6968bb..491c701 100644
--- core/src/components/setup/RecommendedApps.vue
+++ core/src/components/setup/RecommendedApps.vue
@@ -83,7 +83,7 @@ const recommended = {
 		icon: imagePath('core', 'apps/spreed.svg'),
 	},
 	richdocuments: {
-		name: 'Nextcloud Office',
+		name: 'IServ Office',
 		description: t('core', 'Collaborative documents, spreadsheets and presentations, built on Collabora Online.'),
 		icon: imagePath('core', 'apps/richdocuments.svg'),
 	},
