diff --git core/src/components/setup/RecommendedApps.vue core/src/components/setup/RecommendedApps.vue
index f2120c2..a0f9bb2 100644
--- core/src/components/setup/RecommendedApps.vue
+++ core/src/components/setup/RecommendedApps.vue
@@ -84,7 +84,7 @@ const recommended = {
 		icon: imagePath('core', 'apps/spreed.svg'),
 	},
 	richdocuments: {
-		name: 'Nextcloud Office',
+		name: 'IServ Office',
 		description: t('core', 'Collaborative documents, spreadsheets and presentations, built on Collabora Online.'),
 		icon: imagePath('core', 'apps/richdocuments.svg'),
 	},
