diff --git core/src/components/setup/RecommendedApps.vue core/src/components/setup/RecommendedApps.vue
index 472c011..41a2ea5 100644
--- core/src/components/setup/RecommendedApps.vue
+++ core/src/components/setup/RecommendedApps.vue
@@ -101,7 +101,7 @@ const recommended = {
 		icon: imagePath('core', 'apps/spreed.svg'),
 	},
 	richdocuments: {
-		name: 'Nextcloud Office',
+		name: 'IServ Office',
 		description: t('core', 'Collaborative documents, spreadsheets and presentations, built on Collabora Online.'),
 		icon: imagePath('core', 'apps/richdocuments.svg'),
 	},
