diff --git apps/settings/lib/Controller/AuthSettingsController.php apps/settings/lib/Controller/AuthSettingsController.php
index 8652a49..a07cbfd 100644
--- apps/settings/lib/Controller/AuthSettingsController.php
+++ apps/settings/lib/Controller/AuthSettingsController.php
@@ -25,6 +25,7 @@ use OCP\Authentication\Exceptions\InvalidTokenException;
 use OCP\Authentication\Exceptions\WipeTokenException;
 use OCP\Authentication\Token\IToken;
 use OCP\IRequest;
+use OCP\IConfig;
 use OCP\ISession;
 use OCP\IUserSession;
 use OCP\Security\ISecureRandom;
@@ -35,6 +36,9 @@ class AuthSettingsController extends Controller {
 	/** @var IProvider */
 	private $tokenProvider;
 
+	/** @var IConfig */
+	private $config;
+
 	/** @var RemoteWipe */
 	private $remoteWipe;
 
@@ -42,6 +46,7 @@ class AuthSettingsController extends Controller {
 	 * @param string $appName
 	 * @param IRequest $request
 	 * @param IProvider $tokenProvider
+	 * @param IConfig $config
 	 * @param ISession $session
 	 * @param ISecureRandom $random
 	 * @param string|null $userId
@@ -54,6 +59,7 @@ class AuthSettingsController extends Controller {
 		string $appName,
 		IRequest $request,
 		IProvider $tokenProvider,
+		IConfig $config,
 		private ISession $session,
 		private ISecureRandom $random,
 		private ?string $userId,
@@ -64,6 +70,7 @@ class AuthSettingsController extends Controller {
 	) {
 		parent::__construct($appName, $request);
 		$this->tokenProvider = $tokenProvider;
+		$this->config = $config;
 		$this->remoteWipe = $remoteWipe;
 	}
 
@@ -76,6 +83,12 @@ class AuthSettingsController extends Controller {
 	#[NoAdminRequired]
 	#[PasswordConfirmationRequired]
 	public function create($name) {
+
+		// Don't create app auth token if disallowed in configuration.
+		if ($this->config->getSystemValueBool('iserv_disable_app_passwords')) {
+			return $this->getServiceNotAvailableResponse();
+		}
+
 		if ($this->checkAppToken()) {
 			return $this->getServiceNotAvailableResponse();
 		}
