diff --git apps/files_sharing/src/components/SharingEntryLink.vue apps/files_sharing/src/components/SharingEntryLink.vue
index d354c1d..880f203 100644
--- apps/files_sharing/src/components/SharingEntryLink.vue
+++ apps/files_sharing/src/components/SharingEntryLink.vue
@@ -11,17 +11,13 @@
 			class="sharing-entry__avatar" />
 
 		<div class="sharing-entry__summary">
-			<div class="sharing-entry__desc">
+			<div class="sharing-entry__summary__desc" style="align-self: center;">
 				<span class="sharing-entry__title" :title="title">
 					{{ title }}
 				</span>
 				<p v-if="subtitle">
 					{{ subtitle }}
 				</p>
-				<SharingEntryQuickShareSelect v-if="share && share.permissions !== undefined"
-					:share="share"
-					:file-info="fileInfo"
-					@open-sharing-details="openShareDetailsForCustomSettings(share)" />
 			</div>
 
 			<div class="sharing-entry__actions">
@@ -264,7 +260,6 @@ import CheckIcon from 'vue-material-design-icons/CheckBold.vue'
 import CloseIcon from 'vue-material-design-icons/Close.vue'
 import PlusIcon from 'vue-material-design-icons/Plus.vue'
 
-import SharingEntryQuickShareSelect from './SharingEntryQuickShareSelect.vue'
 import ShareExpiryTime from './ShareExpiryTime.vue'
 import SidebarTabExternalActionLegacy from './SidebarTabExternal/SidebarTabExternalActionLegacy.vue'
 
@@ -298,7 +293,6 @@ export default {
 		CheckIcon,
 		CloseIcon,
 		PlusIcon,
-		SharingEntryQuickShareSelect,
 		ShareExpiryTime,
 		SidebarTabExternalActionLegacy,
 	},
