diff --git apps/files_sharing/src/components/SharingEntryLink.vue apps/files_sharing/src/components/SharingEntryLink.vue
index 6a4f52a..a9d28b2 100644
--- apps/files_sharing/src/components/SharingEntryLink.vue
+++ apps/files_sharing/src/components/SharingEntryLink.vue
@@ -11,17 +11,13 @@
 			class="sharing-entry__avatar" />
 
 		<div class="sharing-entry__summary">
-			<div class="sharing-entry__desc">
+			<div class="sharing-entry__summary__desc" style="align-self: center;">
 				<span class="sharing-entry__title" :title="title">
 					{{ title }}
 				</span>
 				<p v-if="subtitle">
 					{{ subtitle }}
 				</p>
-				<SharingEntryQuickShareSelect v-if="share && share.permissions !== undefined"
-					:share="share"
-					:file-info="fileInfo"
-					@open-sharing-details="openShareDetailsForCustomSettings(share)" />
 			</div>
 
 			<!-- clipboard -->
@@ -493,7 +489,7 @@ export default {
 				return true
 			}
 			// Check if either password or expiration date is missing and enforced
-			const isPasswordMissing = this.config.enforcePasswordForPublicLink && !this.share.password
+			const isPasswordMissing = this.config.enforcePasswordForPublicLink && !this.share.newPassword
 			const isExpireDateMissing = this.config.isDefaultExpireDateEnforced && !this.share.expireDate
 
 			return isPasswordMissing || isExpireDateMissing
@@ -623,15 +619,12 @@ export default {
 
 				this.logger.info('Share policy requires a review or has mandated properties (password, expirationDate)...')
 
-				// ELSE, show the pending popovermenu
+				const share = new Share(shareDefaults)
 				// if password default or enforced, pre-fill with random one
 				if (this.config.enableLinkPasswordByDefault || this.config.enforcePasswordForPublicLink) {
-					shareDefaults.password = await GeneratePassword(true)
+					this.$set(share, 'newPassword', await GeneratePassword(true))
 				}
 
-				// create share & close menu
-				const share = new Share(shareDefaults)
-				share.newPassword = share.password
 				const component = await new Promise(resolve => {
 					this.$emit('add:share', share, resolve)
 				})
@@ -695,7 +688,7 @@ export default {
 				const options = {
 					path,
 					shareType: ShareType.Link,
-					password: share.password,
+					password: share.newPassword,
 					expireDate: share.expireDate ?? '',
 					attributes: JSON.stringify(this.fileInfo.shareAttributes),
 					// we do not allow setting the publicUpload
@@ -802,10 +795,8 @@ export default {
 		 * cannot ensure data is up-to-date
 		 */
 		onPasswordDisable() {
-			this.share.password = ''
-
 			// reset password state after sync
-			this.$delete(this.share, 'newPassword')
+			this.$set(this.share, 'newPassword', '')
 
 			// only update if valid share.
 			if (this.share.id) {
