diff --git apps/files_sharing/src/components/SharingEntryLink.vue apps/files_sharing/src/components/SharingEntryLink.vue
index 6a4f52a..c864cdf 100644
--- apps/files_sharing/src/components/SharingEntryLink.vue
+++ apps/files_sharing/src/components/SharingEntryLink.vue
@@ -11,17 +11,13 @@
 			class="sharing-entry__avatar" />
 
 		<div class="sharing-entry__summary">
-			<div class="sharing-entry__desc">
+			<div class="sharing-entry__summary__desc" style="align-self: center;">
 				<span class="sharing-entry__title" :title="title">
 					{{ title }}
 				</span>
 				<p v-if="subtitle">
 					{{ subtitle }}
 				</p>
-				<SharingEntryQuickShareSelect v-if="share && share.permissions !== undefined"
-					:share="share"
-					:file-info="fileInfo"
-					@open-sharing-details="openShareDetailsForCustomSettings(share)" />
 			</div>
 
 			<!-- clipboard -->
