diff --git apps/files_sharing/src/components/SharingInput.vue apps/files_sharing/src/components/SharingInput.vue
index 12429e9..1f3bb68 100644
--- apps/files_sharing/src/components/SharingInput.vue
+++ apps/files_sharing/src/components/SharingInput.vue
@@ -490,7 +490,11 @@ export default {
 					subname = result.extra?.email?.value ?? ''
 					displayName = result.extra?.name?.value ?? displayName
 				} else if (result.value.server) {
-					subname = t('files_sharing', 'on {server}', { server: result.value.server })
+					if (result.shareWithDisplayNameUnique) {
+						subname = result.shareWithDisplayNameUnique
+					} else {
+						subname = t('files_sharing', 'on {server}', { server: result.value.server })
+					}
 				}
 			} else {
 				subname = result.shareWithDescription ?? ''
