diff --git apps/files_sharing/src/components/SharingInput.vue apps/files_sharing/src/components/SharingInput.vue
index 9fee729..d8af383 100644
--- apps/files_sharing/src/components/SharingInput.vue
+++ apps/files_sharing/src/components/SharingInput.vue
@@ -23,6 +23,14 @@
 				{{ search ? noResultText : t('files_sharing', 'No recommendations. Start typing.') }}
 			</template>
 		</NcSelect>
+		<NcCheckboxRadioSwitch
+			v-if="isFederationAppInstalled"
+			type="switch"
+			:checked="onlyLocalUsers"
+			@update:checked="sharingTargets"
+		>
+			{{ t('files_sharing', 'Only local users') }}
+		</NcCheckboxRadioSwitch>
 	</div>
 </template>
 
@@ -40,11 +48,14 @@ import ShareRequests from '../mixins/ShareRequests.js'
 import ShareDetails from '../mixins/ShareDetails.js'
 import { ShareType } from '@nextcloud/sharing'
 
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+
 export default {
 	name: 'SharingInput',
 
 	components: {
 		NcSelect,
+		NcCheckboxRadioSwitch,
 	},
 
 	mixins: [ShareRequests, ShareDetails],
@@ -90,6 +101,7 @@ export default {
 			ShareSearch: OCA.Sharing.ShareSearch.state,
 			suggestions: [],
 			value: null,
+			onlyLocalUsers: false,
 		}
 	},
 
@@ -136,6 +148,11 @@ export default {
 			}
 			return t('files_sharing', 'No elements found.')
 		},
+
+		isFederationAppInstalled() {
+			return OC.appConfig.core.isFederationAppInstalled;
+		},
+
 	},
 
 	mounted() {
@@ -176,8 +193,6 @@ export default {
 			const shareType = [
 				ShareType.User,
 				ShareType.Group,
-				ShareType.Remote,
-				ShareType.RemoteGroup,
 				ShareType.Team,
 				ShareType.Room,
 				ShareType.Guest,
@@ -185,6 +200,11 @@ export default {
 				ShareType.ScienceMesh,
 			]
 
+			if (!this.onlyLocalUsers) {
+				shareType.push(ShareType.Remote);
+				shareType.push(ShareType.RemoteGroup);
+			}
+
 			if (getCapabilities().files_sharing.public.enabled === true) {
 				shareType.push(ShareType.Email)
 			}
@@ -450,6 +470,11 @@ export default {
 				...this.shareTypeToIcon(result.value.shareType),
 			}
 		},
+
+		sharingTargets(value) {
+			this.onlyLocalUsers = value
+			this.asyncFind(this.query)
+		}
 	},
 }
 </script>
