diff --git apps/files_versions/src/components/Version.vue apps/files_versions/src/components/Version.vue
index a126f2a..5bc1e68 100644
--- apps/files_versions/src/components/Version.vue
+++ apps/files_versions/src/components/Version.vue
@@ -272,6 +272,11 @@ export default defineComponent({
 		},
 
 		isDownloadable(): boolean {
+			// Configuration option to disable download buttons
+			if (OC.appConfig.core.disableFileDownloads) {
+				return false
+			}
+
 			if ((this.fileInfo.permissions & Permission.READ) === 0) {
 				return false
 			}
@@ -309,7 +314,11 @@ export default defineComponent({
 
 		click() {
 			if (!this.canView) {
-				window.location.href = this.downloadURL
+				if (this.isDownloadable) {
+					window.location.href = this.downloadURL
+				} else {
+					OC.Notification.showTemporary(t('files_versions', 'This version cannot be viewed.'));
+				}
 				return
 			}
 			this.$emit('click', { version: this.version })
