diff --git apps/files_versions/src/components/Version.vue apps/files_versions/src/components/Version.vue
index c9edbe9..0c99630 100644
--- apps/files_versions/src/components/Version.vue
+++ apps/files_versions/src/components/Version.vue
@@ -268,6 +268,11 @@ export default defineComponent({
 		},
 
 		isDownloadable(): boolean {
+			// Configuration option to disable download buttons
+			if (OC.appConfig.core.disableFileDownloads) {
+				return false
+			}
+
 			if ((this.fileInfo.permissions & Permission.READ) === 0) {
 				return false
 			}
