diff --git lib/private/Template/JSConfigHelper.php lib/private/Template/JSConfigHelper.php
index c3052fc..25d1672 100644
--- lib/private/Template/JSConfigHelper.php
+++ lib/private/Template/JSConfigHelper.php
@@ -246,6 +246,8 @@ class JSConfigHelper {
 					'defaultRemoteExpireDateEnabled' => $defaultRemoteExpireDateEnabled,
 					'defaultRemoteExpireDate' => $defaultRemoteExpireDate,
 					'defaultRemoteExpireDateEnforced' => $defaultRemoteExpireDateEnforced,
+					'disableFileDownloads' => $this->config->getSystemValue('iserv_disable_file_downloads', true),
+					'isFederationAppInstalled' => $this->appManager->isInstalled('federation'),
 				]
 			]),
 			'_theme' => json_encode([
