diff --git lib/private/Template/JSConfigHelper.php lib/private/Template/JSConfigHelper.php
index 7148e4c..4f84f61 100644
--- lib/private/Template/JSConfigHelper.php
+++ lib/private/Template/JSConfigHelper.php
@@ -260,6 +260,8 @@ class JSConfigHelper {
 					'defaultRemoteExpireDateEnabled' => $defaultRemoteExpireDateEnabled,
 					'defaultRemoteExpireDate' => $defaultRemoteExpireDate,
 					'defaultRemoteExpireDateEnforced' => $defaultRemoteExpireDateEnforced,
+					'disableFileDownloads' => $this->config->getSystemValue('iserv_disable_file_downloads', true),
+					'isFederationAppInstalled' => $this->appManager->isInstalled('federation'),
 				]
 			]),
 			"_theme" => json_encode([
