diff --git lib/private/Template/JSConfigHelper.php lib/private/Template/JSConfigHelper.php
index c606041..d81999e 100644
--- lib/private/Template/JSConfigHelper.php
+++ lib/private/Template/JSConfigHelper.php
@@ -253,6 +253,7 @@ class JSConfigHelper {
 					'defaultRemoteExpireDateEnabled' => $defaultRemoteExpireDateEnabled,
 					'defaultRemoteExpireDate' => $defaultRemoteExpireDate,
 					'defaultRemoteExpireDateEnforced' => $defaultRemoteExpireDateEnforced,
+					'disableFileDownloads' => $this->config->getSystemValueBool('iserv_disable_file_downloads', false),
 				]
 			]),
 			'_theme' => json_encode([
