diff --git apps/files_sharing/appinfo/routes.php apps/files_sharing/appinfo/routes.php
index 9d8c576..f9e10eb 100644
--- apps/files_sharing/appinfo/routes.php
+++ apps/files_sharing/appinfo/routes.php
@@ -41,12 +41,6 @@ return [
 			'verb' => 'GET',
 			'root' => '',
 		],
-
-		[
-			'name' => 'externalShares#testRemote',
-			'url' => '/testremote',
-			'verb' => 'GET'
-		],
 		[
 			'name' => 'PublicPreview#getPreview',
 			'url' => '/publicpreview/{token}',
