diff --git apps/richdocuments/lib/Controller/WopiController.php apps/richdocuments/lib/Controller/WopiController.php
index 0e7574001..27c0ffa32 100644
--- apps/richdocuments/lib/Controller/WopiController.php
+++ apps/richdocuments/lib/Controller/WopiController.php
@@ -158,6 +158,8 @@ class WopiController extends Controller {
 		} catch (NoLockProviderException|PreConditionNotMetException) {
 		}
 
+		$downloadDisabled = $wopi->getHideDownload() || $this->config->getSystemValueBool('iserv_disable_file_downloads', false);
+
 		$response = [
 			'BaseFileName' => $file->getName(),
 			'Size' => $file->getSize(),
@@ -177,11 +179,11 @@ class WopiController extends Controller {
 			'EnableShare' => $file->isShareable() && !$isVersion && !$isPublic,
 			'HideUserList' => '',
 			'EnableOwnerTermination' => $wopi->getCanwrite() && !$isPublic,
-			'DisablePrint' => $wopi->getHideDownload(),
-			'DisableExport' => $wopi->getHideDownload(),
-			'DisableCopy' => $wopi->getHideDownload(),
-			'HideExportOption' => $wopi->getHideDownload(),
-			'HidePrintOption' => $wopi->getHideDownload(),
+			'DisablePrint' => $downloadDisabled,
+			'DisableExport' => $downloadDisabled,
+			'DisableCopy' => $downloadDisabled,
+			'HideExportOption' => $downloadDisabled,
+			'HidePrintOption' => $downloadDisabled,
 			'DownloadAsPostMessage' => $wopi->getDirect(),
 			'SupportsLocks' => $this->lockManager->isLockProviderAvailable(),
 			'IsUserLocked' => $this->permissionManager->userIsFeatureLocked($wopi->getEditorUid()),
