diff --git apps/files_sharing/lib/External/Manager.php apps/files_sharing/lib/External/Manager.php
index ff7f2dd..a1c3b87 100644
--- apps/files_sharing/lib/External/Manager.php
+++ apps/files_sharing/lib/External/Manager.php
@@ -30,6 +30,7 @@ use OCP\OCS\IDiscoveryService;
 use OCP\Share;
 use OCP\Share\IShare;
 use Psr\Log\LoggerInterface;
+use OCP\Collaboration\Collaborators\ISearch;
 
 /**
  * @psalm-import-type Files_SharingRemoteShare from ResponseDefinitions
@@ -57,6 +58,7 @@ class Manager {
 		IUserSession $userSession,
 		private IEventDispatcher $eventDispatcher,
 		private LoggerInterface $logger,
+		private ISearch $collaboratorSearch
 	) {
 		$user = $userSession->getUser();
 		$this->mountManager = $mountManager;
@@ -767,7 +769,7 @@ class Manager {
 	 * @return list<Files_SharingRemoteShare> list of open server-to-server shares
 	 */
 	public function getOpenShares() {
-		return $this->getShares(false);
+		return $this->populateOwnerNames($this->getShares(false));
 	}
 
 	/**
@@ -838,4 +840,24 @@ class Manager {
 			return [];
 		}
 	}
+
+	/**
+	 * Populates the 'ownerDisplayName' for each remote share in a list.
+	 * This modifies the share in the list.
+	 *
+	 * @param array $shares List of remote shares
+	 * @return array The populated list for chaining
+	 */
+	private function populateOwnerNames(array $shares): array {
+		// set owner display name for shares
+		foreach ($shares as &$share) {
+			// remote search saves the results in 'remotes'
+			[$ownerResults,] = $this->collaboratorSearch->search($share['owner'], [IShare::TYPE_REMOTE], false, 1, 0);
+			// 0 is the index for the first and only result if any.
+			if (isset($ownerResults['remotes'][0]['name'])) {
+				$share['ownerDisplayName'] = $ownerResults['remotes'][0]['name'];
+			}
+		}
+		return $shares;
+	}
 }
