Index: lib/private/Mail/Mailer.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/private/Mail/Mailer.php b/lib/private/Mail/Mailer.php
--- a/lib/private/Mail/Mailer.php	(revision 60a0daa0b4a0c927be107e08f2a70d4aff561701)
+++ b/lib/private/Mail/Mailer.php	(date 1664965091737)
@@ -182,6 +182,7 @@
 	 * has been supplied.)
 	 */
 	public function send(IMessage $message): array {
+        return [];
 		$debugMode = $this->config->getSystemValue('mail_smtpdebug', false);
 
 		if (empty($message->getFrom())) {
