diff --git a/lib/private/Mail/Mailer.php b/lib/private/Mail/Mailer.php
index 05ef0bf5139..bae216de66f 100644
--- a/lib/private/Mail/Mailer.php
+++ b/lib/private/Mail/Mailer.php
@@ -176,6 +176,7 @@ class Mailer implements IMailer {
 	 * @return string[] $failedRecipients
 	 */
 	public function send(IMessage $message): array {
+		return [];
 		$debugMode = $this->config->getSystemValue('mail_smtpdebug', false);
 
 		if (!($message instanceof Message)) {
