diff --git apps/richdocuments/src/components/AdminSettings.vue apps/richdocuments/src/components/AdminSettings.vue
index 347b660ff..eb43a3553 100644
--- apps/richdocuments/src/components/AdminSettings.vue
+++ apps/richdocuments/src/components/AdminSettings.vue
@@ -460,7 +460,7 @@ export default {
 	},
 	data() {
 		return {
-			productName: loadState('richdocuments', 'productName', 'Nextcloud Office'),
+			productName: loadState('richdocuments', 'productName', 'IServ Office'),
 			hasNextcloudBranding: loadState('richdocuments', 'hasNextcloudBranding', true),
 
 			serverMode: '',
