diff --git apps/richdocuments/src/components/AdminSettings.vue apps/richdocuments/src/components/AdminSettings.vue
index ed845f8d2..8cb3efcf6 100644
--- apps/richdocuments/src/components/AdminSettings.vue
+++ apps/richdocuments/src/components/AdminSettings.vue
@@ -447,7 +447,7 @@ export default {
 	},
 	data() {
 		return {
-			productName: loadState('richdocuments', 'productName', 'Nextcloud Office'),
+			productName: loadState('richdocuments', 'productName', 'IServ Office'),
 			hasNextcloudBranding: loadState('richdocuments', 'hasNextcloudBranding', true),
 
 			serverMode: '',
