diff --git apps/richdocuments/src/components/AdminSettings.vue apps/richdocuments/src/components/AdminSettings.vue
index 253e119ef..d6fd5d170 100644
--- apps/richdocuments/src/components/AdminSettings.vue
+++ apps/richdocuments/src/components/AdminSettings.vue
@@ -483,7 +483,7 @@ export default {
 	},
 	data() {
 		return {
-			productName: loadState('richdocuments', 'productName', 'Nextcloud Office'),
+			productName: loadState('richdocuments', 'productName', 'IServ Office'),
 			hasNextcloudBranding: loadState('richdocuments', 'hasNextcloudBranding', true),
 
 			serverMode: '',
