diff --git apps/federation/lib/Controller/OCSAuthAPIController.php apps/federation/lib/Controller/OCSAuthAPIController.php
index 8412868..24034a3 100644
--- apps/federation/lib/Controller/OCSAuthAPIController.php
+++ apps/federation/lib/Controller/OCSAuthAPIController.php
@@ -7,6 +7,7 @@
  */
 namespace OCA\Federation\Controller;
 
+use OCA\Federation\BackgroundJob\GetSharedSecret;
 use OCA\Federation\DbHandler;
 use OCA\Federation\TrustedServers;
 use OCP\AppFramework\Http;
@@ -129,14 +130,23 @@ class OCSAuthAPIController extends OCSController {
 			throw new OCSForbiddenException();
 		}
 
-		$this->jobList->add(
-			'OCA\Federation\BackgroundJob\GetSharedSecret',
-			[
-				'url' => $url,
-				'token' => $token,
-				'created' => $this->timeFactory->getTime()
-			]
-		);
+		$jobExists = false;
+		foreach ($this->jobList->getJobsIterator(GetSharedSecret::class, null, 0) as $job) {
+			if ($job->getArgument()['url'] === $url && $job->getArgument()['token'] === $token) {
+				$jobExists = true;
+				break;
+			}
+		}
+		if (!$jobExists) {
+			$this->jobList->add(
+				'OCA\Federation\BackgroundJob\GetSharedSecret',
+				[
+					'url' => $url,
+					'token' => $token,
+					'created' => $this->timeFactory->getTime()
+				]
+			);
+		}
 
 		return new DataResponse();
 	}
@@ -162,7 +172,8 @@ class OCSAuthAPIController extends OCSController {
 		}
 
 		if ($this->isValidToken($url, $token) === false) {
-			$this->throttler->registerAttempt('federationSharedSecret', $this->request->getRemoteAddress());
+			usleep(1000); // Make brute forcing tokens harder, but do not block servers altogether
+			// $this->throttler->registerAttempt('federationSharedSecret', $this->request->getRemoteAddress());
 			$expectedToken = $this->dbHandler->getToken($url);
 			$this->logger->error(
 				'remote server (' . $url . ') didn\'t send a valid token (got "' . $token . '" but expected "'. $expectedToken . '") while getting shared secret',
