diff --git apps/settings/composer/composer/autoload_classmap.php apps/settings/composer/composer/autoload_classmap.php
index f856bd7..2ba7407 100644
--- apps/settings/composer/composer/autoload_classmap.php
+++ apps/settings/composer/composer/autoload_classmap.php
@@ -19,6 +19,7 @@ return array(
     'OCA\\Settings\\Command\\AdminDelegation\\Add' => $baseDir . '/../lib/Command/AdminDelegation/Add.php',
     'OCA\\Settings\\Command\\AdminDelegation\\Remove' => $baseDir . '/../lib/Command/AdminDelegation/Remove.php',
     'OCA\\Settings\\Command\\AdminDelegation\\Show' => $baseDir . '/../lib/Command/AdminDelegation/Show.php',
+    'OCA\\Settings\\Command\\AdminDelegation\\Check' => $baseDir . '/../lib/Command/AdminDelegation/Check.php',
     'OCA\\Settings\\ConfigLexicon' => $baseDir . '/../lib/ConfigLexicon.php',
     'OCA\\Settings\\Controller\\AISettingsController' => $baseDir . '/../lib/Controller/AISettingsController.php',
     'OCA\\Settings\\Controller\\AdminSettingsController' => $baseDir . '/../lib/Controller/AdminSettingsController.php',
