diff --git apps/files_sharing/l10n/de_DE.js apps/files_sharing/l10n/de_DE.js
index 56ae703..7972b7e 100644
--- apps/files_sharing/l10n/de_DE.js
+++ apps/files_sharing/l10n/de_DE.js
@@ -1,6 +1,7 @@
 OC.L10N.register(
     "files_sharing",
     {
+	"Only local users": "Nur lokale Benutzer",
     "File shares" : "Dateifreigaben",
     "Downloaded via public link" : "Über den öffentlichen Link heruntergeladen",
     "Downloaded by {email}" : "Heruntergeladen von {email}",
