diff --git apps/files_sharing/l10n/de_DE.js apps/files_sharing/l10n/de_DE.js
index 2c75939..861a076 100644
--- apps/files_sharing/l10n/de_DE.js
+++ apps/files_sharing/l10n/de_DE.js
@@ -388,6 +388,7 @@ OC.L10N.register(
     "the item was removed" : "Das Element wurde entfernt",
     "the link expired" : "Der Link ist abgelaufen",
     "sharing is disabled" : "Teilen ist deaktiviert",
-    "For more info, please ask the person who sent this link." : "Um nähere Informationen zu erhalten, wenden Sie sich bitte an die Person, die Ihnen diesen Link geschickt hat."
+    "For more info, please ask the person who sent this link." : "Um nähere Informationen zu erhalten, wenden Sie sich bitte an die Person, die Ihnen diesen Link geschickt hat.",
+	"Only local users": "Nur lokale Benutzer"
 },
 "nplurals=2; plural=(n != 1);");
