diff --git apps/files/src/components/FileEntry/FileEntryActions.vue apps/files/src/components/FileEntry/FileEntryActions.vue
index 9ed8de1..7afee04 100644
--- apps/files/src/components/FileEntry/FileEntryActions.vue
+++ apps/files/src/components/FileEntry/FileEntryActions.vue
@@ -243,11 +243,11 @@ export default defineComponent({
 		},
 
 		renderedNonDestructiveActions() {
-			return this.enabledMenuActions.filter(action => !action.destructive)
+			return this.menuActions.filter(action => !action.destructive)
 		},
 
 		renderedDestructiveActions() {
-			return this.enabledMenuActions.filter(action => action.destructive)
+			return this.menuActions.filter(action => action.destructive)
 		},
 
 		openedMenu: {
@@ -267,6 +267,18 @@ export default defineComponent({
 		getBoundariesElement() {
 			return document.querySelector('.app-content > .files-list')
 		},
+
+		// Configuration option to disable download buttons
+		enabledMenuActionsWithoutDownload() {
+			// Filter out download from enabledMenuActions
+			return this.enabledMenuActions.filter(action => action.id !== 'download')
+		},
+
+		// Configuration option to disable download buttons
+		menuActions() {
+			// Checks deactivation of the file download
+			return OC.appConfig.core.disableFileDownloads ? this.enabledMenuActionsWithoutDownload : this.enabledMenuActions
+		},
 	},
 
 	watch: {
