diff --git apps/files/src/components/FileEntry/FileEntryActions.vue apps/files/src/components/FileEntry/FileEntryActions.vue
index 50642d0..657a88c 100644
--- apps/files/src/components/FileEntry/FileEntryActions.vue
+++ apps/files/src/components/FileEntry/FileEntryActions.vue
@@ -26,7 +26,7 @@
 			@close="onMenuClose"
 			@closed="onMenuClosed">
 			<!-- Default actions list-->
-			<NcActionButton v-for="action, index in enabledMenuActions"
+			<NcActionButton v-for="action, index in menuActions"
 				:key="action.id"
 				:ref="`action-${action.id}`"
 				class="files-list__row-action"
@@ -240,6 +240,18 @@ export default defineComponent({
 		getBoundariesElement() {
 			return document.querySelector('.app-content > .files-list')
 		},
+
+		// Configuration option to disable download buttons
+		enabledMenuActionsWithoutDownload() {
+			// Filter out download from enabledMenuActions
+			return this.enabledMenuActions.filter(action => action.id !== 'download')
+		},
+
+		// Configuration option to disable download buttons
+		menuActions() {
+			// Checks deactivation of the file download
+			return OC.appConfig.core.disableFileDownloads ? this.enabledMenuActionsWithoutDownload : this.enabledMenuActions
+		},
 	},
 
 	watch: {
