diff --git apps/viewer/src/views/Viewer.vue apps/viewer/src/views/Viewer.vue
index 0dd20ee..5adcb77 100644
--- apps/viewer/src/views/Viewer.vue
+++ apps/viewer/src/views/Viewer.vue
@@ -76,7 +76,7 @@
 				@click="showSidebar">
 				{{ t('viewer', 'Open sidebar') }}
 			</NcActionButton>
-			<NcActionButton v-if="canDownload"
+			<NcActionButton v-if="enabledDownloads && canDownload"
 				:close-after-click="true"
 				@click="onDownload">
 				<template #icon>
@@ -432,6 +432,16 @@ export default defineComponent({
 				return false
 			}
 		},
+
+		/**
+		 * Configuration option to disable download buttons
+		 *
+		 * @return {boolean}
+		 */
+		enabledDownloads() {
+			return !OC.appConfig.core.disableFileDownloads;
+
+		},
 	},
 
 	watch: {
