diff --git apps/viewer/src/views/Viewer.vue apps/viewer/src/views/Viewer.vue
index 2026033..5ca27d2 100644
--- apps/viewer/src/views/Viewer.vue
+++ apps/viewer/src/views/Viewer.vue
@@ -92,7 +92,7 @@
 				@click="showSidebar">
 				{{ t('viewer', 'Open sidebar') }}
 			</NcActionButton>
-			<NcActionLink v-if="canDownload"
+			<NcActionLink v-if="enabledDownloads && canDownload"
 				:download="currentFile.basename"
 				:close-after-click="true"
 				:href="downloadPath">
@@ -441,6 +441,16 @@ export default {
 				return false
 			}
 		},
+
+		/**
+		 * Configuration option to disable download buttons
+		 *
+		 * @return {boolean}
+		 */
+		enabledDownloads() {
+			return !OC.appConfig.core.disableFileDownloads;
+
+		},
 	},
 
 	watch: {
