diff --git apps/files_linkeditor/src/views/Editor.svelte apps/files_linkeditor/src/views/Editor.svelte
index 6c571cb..b47af4a 100644
--- apps/files_linkeditor/src/views/Editor.svelte
+++ apps/files_linkeditor/src/views/Editor.svelte
@@ -55,17 +55,6 @@
 					/>
 				</label>
 				<input type="checkbox" bind:checked={file.sameWindow} id="linkeditor_sameWindow" class="checkbox" />
-				<label for="linkeditor_sameWindow" class="space-top">{t("files_linkeditor", "Open in same window")}</label>
-				<input
-					type="checkbox"
-					disabled={!file.sameWindow}
-					bind:checked={file.skipConfirmation}
-					id="linkeditor_skipConfirmation"
-					class="checkbox"
-				/>
-				<label for="linkeditor_skipConfirmation">
-					{t("files_linkeditor", "Skip confirmation dialog before open (has to open in same window)")}
-				</label>
 			{/if}
 		</div>
 		<div class="oc-dialog-buttonrow onebutton urlvisit">
