diff --git apps/files_sharing/src/views/SharingDetailsTab.vue apps/files_sharing/src/views/SharingDetailsTab.vue
index c563aa0..c3a5b4e 100644
--- apps/files_sharing/src/views/SharingDetailsTab.vue
+++ apps/files_sharing/src/views/SharingDetailsTab.vue
@@ -36,6 +36,7 @@
 						</template>
 					</NcCheckboxRadioSwitch>
 					<NcCheckboxRadioSwitch :button-variant="true"
+					    v-if="canEditFileType"
 						data-cy-files-sharing-share-permissions-bundle="upload-edit"
 						:checked.sync="sharingPermission"
 						:value="allPermissions"
@@ -149,7 +150,7 @@
 						:placeholder="t('files_sharing', 'Expiration date')"
 						type="date"
 						@input="onExpirationChange" />
-					<NcCheckboxRadioSwitch v-if="isPublicShare"
+					<NcCheckboxRadioSwitch v-if="isPublicShare && !downloadDisabled"
 						:disabled="canChangeHideDownload"
 						:checked.sync="share.hideDownload"
 						@update:checked="queueUpdate('hideDownload')">
@@ -191,7 +192,7 @@
 							data-cy-files-sharing-share-permissions-checkbox="create">
 							{{ t('files_sharing', 'Create') }}
 						</NcCheckboxRadioSwitch>
-						<NcCheckboxRadioSwitch :disabled="!canSetEdit"
+						<NcCheckboxRadioSwitch :disabled="!canSetEdit || !canEditFileType"
 							:checked.sync="canEdit"
 							data-cy-files-sharing-share-permissions-checkbox="update">
 							{{ t('files_sharing', 'Edit') }}
@@ -345,6 +346,39 @@ export default {
 	},
 
 	computed: {
+		canEditFileType () {
+			// Taken from https://github.com/nextcloud/text/blob/81f3256c03ebf2692e803486c0dcb6a8b442ce67/lib/DirectEditing/TextDirectEditor.php#L86
+			const supportedMimeTypes = [
+				'text/markdown',
+				'text/plain',
+				'application/cmd',
+				'application/x-empty',
+				'application/x-msdos-program',
+				'application/javascript',
+				'application/json',
+				'application/x-perl',
+				'application/x-php',
+				'application/x-tex',
+				'application/xml',
+				'application/yaml',
+				'text/css',
+				'text/csv',
+				'text/html',
+				'text/org',
+				'text/x-c',
+				'text/x-c++src',
+				'text/x-h',
+				'text/x-java-source',
+				'text/x-ldif',
+				'text/x-python',
+				'text/x-shellscript',
+				'application/pdf',
+			];
+			return this.isFolder || supportedMimeTypes.includes(this.fileInfo.mimetype?.toLowerCase());
+		},
+		downloadDisabled() {
+			return OC.appConfig.core.disableFileDownloads === true;
+		},
 		title() {
 			switch (this.share.type) {
 			case ShareType.User:
@@ -383,7 +417,7 @@ export default {
 		 */
 		canEdit: {
 			get() {
-				return this.share.hasUpdatePermission
+				return this.share.hasUpdatePermission && this.canEditFileType
 			},
 			set(checked) {
 				this.updateAtomicPermissions({ isEditChecked: checked })
@@ -427,7 +461,7 @@ export default {
 		 */
 		canDownload: {
 			get() {
-				return this.share.attributes.find(attr => attr.key === 'download')?.value || false
+				return !this.downloadDisabled && (this.share.attributes.find(attr => attr.key === 'download')?.value || false);
 			},
 			set(checked) {
 				this.setShareAttribute('permissions', 'download', checked)
