diff --git apps/files_sharing/src/views/SharingDetailsTab.vue apps/files_sharing/src/views/SharingDetailsTab.vue
index b0cb044..fe6b66d 100644
--- apps/files_sharing/src/views/SharingDetailsTab.vue
+++ apps/files_sharing/src/views/SharingDetailsTab.vue
@@ -33,16 +33,14 @@
 					</NcCheckboxRadioSwitch>
 					<NcCheckboxRadioSwitch :button-variant="true"
 						data-cy-files-sharing-share-permissions-bundle="upload-edit"
+						v-if="canEditFileType && !isPublicShare"
 						:checked.sync="sharingPermission"
 						:value="bundledPermissions.ALL.toString()"
 						name="sharing_permission_radio"
 						type="radio"
 						button-variant-grouped="vertical"
 						@update:checked="toggleCustomPermissions">
-						<template v-if="allowsFileDrop">
-							{{ t('files_sharing', 'Allow upload and editing') }}
-						</template>
-						<template v-else>
+						<template>
 							{{ t('files_sharing', 'Allow editing') }}
 						</template>
 						<template #icon>
@@ -165,10 +163,10 @@
 						</label>
 						<textarea id="share-note-textarea" :value="share.note" @input="share.note = $event.target.value" />
 					</template>
-					<NcCheckboxRadioSwitch :checked.sync="setCustomPermissions">
+					<NcCheckboxRadioSwitch v-if="!isPublicShare" :checked.sync="setCustomPermissions">
 						{{ t('files_sharing', 'Custom permissions') }}
 					</NcCheckboxRadioSwitch>
-					<section v-if="setCustomPermissions" class="custom-permissions-group">
+					<section v-if="setCustomPermissions && !isPublicShare" class="custom-permissions-group">
 						<NcCheckboxRadioSwitch :disabled="!canRemoveReadPermission"
 							:checked.sync="hasRead"
 							data-cy-files-sharing-share-permissions-checkbox="read">
@@ -180,7 +178,7 @@
 							data-cy-files-sharing-share-permissions-checkbox="create">
 							{{ t('files_sharing', 'Create') }}
 						</NcCheckboxRadioSwitch>
-						<NcCheckboxRadioSwitch :disabled="!canSetEdit"
+						<NcCheckboxRadioSwitch :disabled="!canSetEdit || !canEditFileType"
 							:checked.sync="canEdit"
 							data-cy-files-sharing-share-permissions-checkbox="update">
 							{{ t('files_sharing', 'Edit') }}
@@ -325,6 +323,35 @@ export default {
 	},
 
 	computed: {
+        canEditFileType () {
+            // Taken from https://github.com/nextcloud/text/blob/0320046ffe8b853e8e329a9b865cdf3633228509/lib/DirectEditing/TextDirectEditor.php#L83
+            const supportedMimeTypes = [
+                'text/markdown',
+                'text/plain',
+                'application/cmd',
+                'application/x-empty',
+                'application/x-msdos-program',
+                'application/javascript',
+                'application/json',
+                'application/x-perl',
+                'application/x-php',
+                'application/x-tex',
+                'application/xml',
+                'application/yaml',
+                'text/css',
+                'text/csv',
+                'text/html',
+                'text/org',
+                'text/x-c',
+                'text/x-c++src',
+                'text/x-h',
+                'text/x-java-source',
+                'text/x-ldif',
+                'text/x-python',
+                'text/x-shellscript'
+            ];
+            return this.isFolder || supportedMimeTypes.includes(this.fileInfo.mimetype?.toLowerCase());
+        },
 		title() {
 			switch (this.share.type) {
 			case this.SHARE_TYPES.SHARE_TYPE_USER:
