diff --git apps/files_sharing/src/views/SharingDetailsTab.vue apps/files_sharing/src/views/SharingDetailsTab.vue
index b0cb044..fe6b66d 100644
--- apps/files_sharing/src/views/SharingDetailsTab.vue
+++ apps/files_sharing/src/views/SharingDetailsTab.vue
@@ -32,6 +32,7 @@
 						</template>
 					</NcCheckboxRadioSwitch>
 					<NcCheckboxRadioSwitch :button-variant="true"
+                        v-if="canEditFileType"
 						data-cy-files-sharing-share-permissions-bundle="upload-edit"
 						:checked.sync="sharingPermission"
 						:value="bundledPermissions.ALL.toString()"
@@ -180,7 +181,7 @@
 							data-cy-files-sharing-share-permissions-checkbox="create">
 							{{ t('files_sharing', 'Create') }}
 						</NcCheckboxRadioSwitch>
-						<NcCheckboxRadioSwitch :disabled="!canSetEdit"
+						<NcCheckboxRadioSwitch :disabled="!canSetEdit || !canEditFileType"
 							:checked.sync="canEdit"
 							data-cy-files-sharing-share-permissions-checkbox="update">
 							{{ t('files_sharing', 'Edit') }}
@@ -325,6 +326,36 @@ export default {
 	},

 	computed: {
+        canEditFileType () {
+            // Taken from https://github.com/nextcloud/text/blob/81f3256c03ebf2692e803486c0dcb6a8b442ce67/lib/DirectEditing/TextDirectEditor.php#L86
+            const supportedMimeTypes = [
+                'text/markdown',
+                'text/plain',
+                'application/cmd',
+                'application/x-empty',
+                'application/x-msdos-program',
+                'application/javascript',
+                'application/json',
+                'application/x-perl',
+                'application/x-php',
+                'application/x-tex',
+                'application/xml',
+                'application/yaml',
+                'text/css',
+                'text/csv',
+                'text/html',
+                'text/org',
+                'text/x-c',
+                'text/x-c++src',
+                'text/x-h',
+                'text/x-java-source',
+                'text/x-ldif',
+                'text/x-python',
+                'text/x-shellscript',
+                'application/pdf',
+            ];
+            return this.isFolder || supportedMimeTypes.includes(this.fileInfo.mimetype?.toLowerCase());
+        },
 		title() {
 			switch (this.share.type) {
 			case this.SHARE_TYPES.SHARE_TYPE_USER:
@@ -360,7 +391,7 @@ export default {
 		 */
 		canEdit: {
 			get() {
-				return this.share.hasUpdatePermission
+				return this.share.hasUpdatePermission && this.canEditFileType
 			},
 			set(checked) {
 				this.updateAtomicPermissions({ isEditChecked: checked })
