--- apps/federation/composer/composer/autoload_classmap.php	2024-02-16 09:54:49.856156471 +0100
+++ apps/federation/composer/composer/autoload_classmap_patched.php	2024-02-29 08:08:08.478623496 +0100
@@ -11,6 +11,9 @@
     'OCA\\Federation\\BackgroundJob\\GetSharedSecret' => $baseDir . '/../lib/BackgroundJob/GetSharedSecret.php',
     'OCA\\Federation\\BackgroundJob\\RequestSharedSecret' => $baseDir . '/../lib/BackgroundJob/RequestSharedSecret.php',
     'OCA\\Federation\\Command\\SyncFederationAddressBooks' => $baseDir . '/../lib/Command/SyncFederationAddressBooks.php',
+	'OCA\\Federation\\Command\\AddTrustedServer' => $baseDir . '/../lib/Command/AddTrustedServer.php',
+	'OCA\\Federation\\Command\\ListTrustedServer' => $baseDir . '/../lib/Command/ListTrustedServer.php',
+	'OCA\\Federation\\Command\\RemoveTrustedServer' => $baseDir . '/../lib/Command/RemoveTrustedServer.php',
     'OCA\\Federation\\Controller\\OCSAuthAPIController' => $baseDir . '/../lib/Controller/OCSAuthAPIController.php',
     'OCA\\Federation\\Controller\\SettingsController' => $baseDir . '/../lib/Controller/SettingsController.php',
     'OCA\\Federation\\DAV\\FedAuth' => $baseDir . '/../lib/DAV/FedAuth.php',
