diff --git apps/files/src/components/FilesListTableHeaderActions.vue apps/files/src/components/FilesListTableHeaderActions.vue
index 6f02671..67e176e 100644
--- apps/files/src/components/FilesListTableHeaderActions.vue
+++ apps/files/src/components/FilesListTableHeaderActions.vue
@@ -14,7 +14,7 @@
 			:open.sync="openedMenu"
 			@close="openedSubmenu = null">
 			<!-- Default actions list-->
-			<NcActionButton v-for="action in enabledMenuActions"
+			<NcActionButton v-for="action in customIServActions"
 				:key="action.id"
 				:ref="`action-batch-${action.id}`"
 				:class="{
@@ -256,6 +256,18 @@ export default defineComponent({
 			}
 			return 3
 		},
+
+		// Configuration option to disable download buttons
+		enabledMenuActionsWithoutDownload() {
+			// Filter out download from enabledMenuActions
+			return this.enabledMenuActions.filter(action => action.id !== 'download')
+		},
+
+		// Configuration option to disable download buttons
+		customIServActions() {
+			// Checks deactivation of the file download
+			return OC.appConfig.core.disableFileDownloads ? this.enabledMenuActionsWithoutDownload : this.enabledMenuActions
+		},
 	},
 
 	methods: {
