diff --git lib/private/AppFramework/DependencyInjection/DIContainer.php lib/private/AppFramework/DependencyInjection/DIContainer.php
index 76261fe..cf1103f 100644
--- lib/private/AppFramework/DependencyInjection/DIContainer.php
+++ lib/private/AppFramework/DependencyInjection/DIContainer.php
@@ -182,6 +182,11 @@ class DIContainer extends SimpleContainer implements IAppContainer {
 
 			$dispatcher = new MiddlewareDispatcher();
 
+			// execute early before everything
+			$dispatcher->registerMiddleware(
+				$c->get(\OC\Core\Middleware\BlockClientMiddleware::class)
+			);
+
 			$dispatcher->registerMiddleware($c->get(CompressionMiddleware::class));
 			$dispatcher->registerMiddleware($c->get(NotModifiedMiddleware::class));
 			$dispatcher->registerMiddleware($c->get(ReloadExecutionMiddleware::class));
